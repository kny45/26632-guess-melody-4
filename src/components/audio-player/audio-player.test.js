import React from "react";
import renderer from "react-test-renderer";
import AudioPlayer from "./audio-player.jsx";

describe(`Render AudioPlayer`, () => {
  const mock = `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg`;

  it(`AudioPlayer is rendered correctly with play`, () => {
    const tree = renderer.create(
        <AudioPlayer
          isPlaying = {true}
          src = {mock}
          onPlayButtonClick = {() => {}}
        />, {
          createNodeMock: () => {
            return {};
          }
        }
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`AudioPlayer is rendered correctly with pause`, () => {
    const tree = renderer.create(
        <AudioPlayer
          isPlaying = {false}
          src = {mock}
          onPlayButtonClick = {() => {}}
        />, {
          createNodeMock: () => {
            return {};
          }
        }
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
