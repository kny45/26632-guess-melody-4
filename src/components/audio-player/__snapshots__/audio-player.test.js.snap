// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render AudioPlayer AudioPlayer is rendered correctly with pause 1`] = `
Array [
  <button
    className="track__button track__button--play"
    disabled={true}
    onClick={[Function]}
    type="button"
  />,
  <div
    className="track__status"
  >
    <audio />
  </div>,
]
`;

exports[`Render AudioPlayer AudioPlayer is rendered correctly with play 1`] = `
Array [
  <button
    className="track__button track__button--pause"
    disabled={true}
    onClick={[Function]}
    type="button"
  />,
  <div
    className="track__status"
  >
    <audio />
  </div>,
]
`;
